Tests/configure_and_test_integration_instances.py

```diff
@@ -1290,136 +1566,67 @@ def get_non_added_packs_ids(build: Build):
-    """
-    tests_for_iteration = get_tests(build)
-    modified_module_instances, new_module_instances = configure_server_instances(build,
-                                                                                 tests_for_iteration,
-                                                                                 new_integrations,
-                                                                                 modified_integrations)
-    successful_tests_pre, failed_tests_pre = instance_testing(build, modified_module_instances, pre_update=True)
-    return modified_module_instances, new_module_instances, failed_tests_pre, successful_tests_pre
-
-
-def install_packs_pre_update(build: Build) -> bool:
-    """
-    Install packs on server according to server version
-    Args:
-        build: A build object
-
-    Returns:
-        A boolean that indicates whether the installation was successful or not
-    """
-    installed_content_packs_successfully = False
-    if Version(build.server_numeric_version) >= Version('6.0.0'):
-        if build.is_nightly:
-            install_nightly_pack(build)
-            installed_content_packs_successfully = True
-        else:
-            if not build.is_private:
-                pack_ids = get_non_added_packs_ids(build)
-                installed_content_packs_successfully = install_packs(build, pack_ids=pack_ids)
+def create_build_object() -> Build:
+    options = options_handler()
+    logging.info(f'Build type: {options.build_object_type}')
+    if options.build_object_type == XSOAR_BUILD_TYPE:
+        return XSOARBuild(options)
+    elif options.build_object_type == XSIAM_BUILD_TYPE:
+        return XSIAMBuild(options)
     else:
-        installed_content_packs_successfully = True
-    return installed_content_packs_successfully
+        raise Exception(f"Wrong Build object type {options.build_object_type}.")
 
 
 def main():
+    """
+    This step in the build doing different things for branch build and nightly.
+    The flow for custom branch build is:
+        1. Add server config and restart servers (only in xsoar).
+        2. Disable all enabled integrations.
+        3. Finds only modified (not new) packs and install them, same version as in production.
+            (before the update in this branch).
+        4. Compares master to commit_sha and return two lists - new integrations and modified in the current branch.
+        5. Configures integration instances (same version as in production) for the modified packs
+            and runs `test-module` (pre-update).
+        6. Changes marketplace bucket to the new one that was created in create-instances workflow.
+        7. Installs all (new and modified) packs from current branch.
+        8. After updating packs from branch, runs `test-module` for both new and modified integrations,
+            to check that modified integrations was not broken. (post-ipdate).
+        9. Prints results.
```

Comment:
Shouldn't the uninstalling step be mentioned here too?

Reply:
No, since It is not happening in this step.